fuel-tx/src/tx_pointer.rs

```diff
@@ -70,12 +83,28 @@ impl fmt::Display for TxPointer {
     }
 }
 
+#[cfg(feature = "u32-tx-pointer")]
+impl fmt::LowerHex for TxPointer {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{:08x}{:08x}", self.block_height, self.tx_index)
+    }
+}
+
+#[cfg(not(feature = "u32-tx-pointer"))]
 impl fmt::LowerHex for TxPointer {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
         write!(f, "{:08x}{:04x}", self.block_height, self.tx_index)
     }
 }
 
+#[cfg(feature = "u32-tx-pointer")]
+impl fmt::UpperHex for TxPointer {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{:08X}{:08X}", self.block_height, self.tx_index)
+    }
+}
```

Comment:
It would be nice if we can make it to use old representation in the case of small number and big one in the case of big numbers=D In this case it will be backward compatible.

Reply:
Done. I also updated `from_str` to read old version then.